app/uk/gov/hmrc/apinotificationpull/logging/NotificationLogger.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apinotificationpull.logging
+
+import com.google.inject.Inject
+import javax.inject.Singleton
+import uk.gov.hmrc.apinotificationpull.logging.LoggingHelper.formatWithHeaders
+import uk.gov.hmrc.apinotificationpull.model.SeqOfHeader
+import uk.gov.hmrc.customs.api.common.logging.CdsLogger
+
+@Singleton
+class NotificationLogger @Inject()(logger: CdsLogger) {
+
```

Comment:
I would consider making 2nd parameter `headers: => SeqOfHeader` implicit.
You could also create an implicit function or class that takes a HeaderCarrier as input and returns a `headers: => SeqOfHeader` . This should help to cut down on some of the repetition